logger/common.go

```diff
@@ -127,26 +142,27 @@ func (l *Logger) Start(
 	if l.Stdout == nil || l.Stderr == nil {
 		return errors.New("no stdout/stderr pipe opened")
 	}
+	pipeNameToPipe := map[string]io.Reader{
+		sourceSTDOUT: l.Stdout,
+		sourceSTDERR: l.Stderr,
+	}
 
 	errGroup, ctx := errgroup.WithContext(ctx)
-	errGroup.Go(func() error {
-		err := l.sendLogs(ctx, l.Stdout, sourceSTDOUT, uid, gid, cleanupTime)
-		if err != nil {
-			stdoutErr := errors.Wrapf(err, "failed to send logs from pipe %s", sourceSTDOUT)
-			debug.SendEventsToJournal(DaemonName, stdoutErr.Error(), journal.PriErr, 1)
-			return stdoutErr
-		}
-		return nil
-	})
-	errGroup.Go(func() error {
-		err := l.sendLogs(ctx, l.Stderr, sourceSTDERR, uid, gid, cleanupTime)
-		if err != nil {
-			stderrErr := errors.Wrapf(err, "failed to send logs from pipe %s", sourceSTDERR)
-			debug.SendEventsToJournal(DaemonName, stderrErr.Error(), journal.PriErr, 1)
-			return stderrErr
-		}
-		return nil
-	})
+	for source, pipe := range pipeNameToPipe {
+		// Copy source and pipe to a new variable
+		source := source
```

Comment:
can we use different var names here? Makes it easy to grok.

Reply:
Sure, will update with two different names here